src/metabase/models/serialization.clj

```diff
@@ -1482,6 +1506,45 @@
   (set/union (viz-click-behavior-descendants  viz)
              (viz-column-settings-descendants viz)))
 
+;;; Common transformers
+
+(defn fk "Export Foreign Key" [model & [field-name]]
+  (cond
+    ;; this `::fk` is used in tests to determine that foreign keys are handled
+    (= model :model/User)  ^::fk [*export-user* *import-user*]
+    (= model :model/Table) ^::fk [*export-table-fk* *import-table-fk*]
+    (= model :model/Field) ^::fk [*export-field-fk* *import-field-fk*]
+    field-name             ^::fk [#(*export-fk-keyed* % model field-name) #(*import-fk-keyed* % model field-name)]
+    :else                  ^::fk [#(*export-fk* % model) #(*import-fk* % model)]))
+
+(defn nested "Nested entities" [model backward-fk opts]
+  (let [model-name (name model)
+        sorter     (:sort-by opts :created_at)
+        key-field  (:key-field opts :entity_id)]
+    ^::post
+    [(fn [data]
+       ;; if this looks weird, see :series hydration for DashboardCard, it supplies Cards while we need to serialize
+       ;; DashboardCardSeries instances
+       (let [entities (if (t2/instance-of? model (first data))
+                        data
+                        (t2/select model backward-fk (:id *current*)))]
+         (->> (mapv #(extract-one model-name opts %) entities)
+              (sort-by sorter))))
+     (fn [lst]
+       (let [parent-id (:id *current*)
+             loaded    (for [ingested lst
+                             ;; it's not always entity-id though, not for DashcardCardSeries
+                             :let     [entity-id (get ingested key-field)]]
```

Comment:
Do nested entities need an entity ID? Nested entities cannot be loaded independently by `serdes/load-one!` or listed as a dependency by `serdes/dependencies`. Instead they are usually linked by foreign key relationships to another entity that can be identified with an `entity_id`.

Reply:
Well, they are working like that right now, tabs and dashcards are serialized via entity_id.

Reply:
yep sorry, I now see how `entity_id` is used now for dashcards and tabs. It is useful for identifying when to replace vs update the local entities